viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
+        sum += user_data["watched"][i]["rating"]
+        average = sum  / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+    index = 0
+    genres = []
+    least_popular = []
+    most_popular = []
+    
+    if user_data == {"watched":[]}:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            index +=1
+            genres.append(user_data["watched"][index-1]["genre"])
+        for item in genres:
+            if item not in least_popular:
+                least_popular.append(item)
+            else:
+                most_popular.append(item)
+        return (most_popular[0])
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data): 
+    unique = []
+    combined_movies = [] 
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    combined_movies = users_movies + friends_movies
+
+    for item in combined_movies:
+        if item not in friends_movies:
+            unique.append(item)
+    return unique
+    
+def get_friends_unique_watched(user_data):
+    left_to_watch_by_user = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in left_to_watch_by_user:
+                left_to_watch_by_user.append(item)
+    return left_to_watch_by_user
+
+# ------------- WAVE 4 -------------------
+
+def get_available_recs(user_data):
+    watched_movies = []
+
+    for movie in user_data["watched"]:
+        watched_movies.append(movie["title"])
+
+    recommended = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in watched_movies:
+                if movie["host"] in user_data["subscriptions"]:
+                    recommended.append(movie)
+    return recommended
 
-# -----------------------------------------
 # ------------- WAVE 5 --------------------
-# -----------------------------------------
+
+def get_new_rec_by_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return user_data["watched"]
+
+    genre_watched_by_user = []
+    for movie in user_data["watched"]:
+        genre_watched_by_user.append(movie["genre"])
+
+    most_frequent = max(set(genre_watched_by_user), key= genre_watched_by_user.count)

```

Comment:
This is a solution to `get_most_watched_genre`.  How would you solve this without the max function?